src/cache.h

```diff
@@ -270,211 +330,541 @@ cache_ctxt_init(cache            *cc,
+ *
+ * The pin value is reference counted, so multiple threads may pin
+ * and unpin, and the page is only eligible for eviction when no threads
+ * have it pinned.
+ *
+ *----------------------------------------------------------------------
+ */
 static inline void
 cache_unpin(cache *cc, page_handle *page)
 {
    return cc->ops->page_unpin(cc, page);
 }
 
+/*
+ *-----------------------------------------------------------------------------
+ * cache_page_sync
+ *
+ * Asynchronously writes the page back to disk. Currently there is no way to
+ * check when the writeback has completed,
+ * XXX thus this call is only useful to prepare a page for eviction?
+ *
+ *-----------------------------------------------------------------------------
+ */
 static inline void
 cache_page_sync(cache *cc, page_handle *page, bool is_blocking, page_type type)
 {
    return cc->ops->page_sync(cc, page, is_blocking, type);
 }
 
+/*
+ *-----------------------------------------------------------------------------
+ * cache_extent_sync
+ *
+ * Asynchronously syncs the extent beginning at addr.
+ *
+ * *pages_outstanding is immediately incremented by the number of pages
+ * issued for writeback (the non-clean pages of the extent); as writebacks
+ * complete, *pages_outstanding is decremented. The caller may use the
+ * XXX atomically-read? Because it's an aligned uint64?
+ * value of *pages_outstanding to determine when the extent writeback is
+ * complete.
+ *
+ * All pages in the extent must be clean or cleanable.
+ * XXX What other states might there be, and what would happen if you
+ * called this with a page in such a state?
+ * XXX should we copy this remark from flush?
+ * XXX Asserts that there are no pins, read locks, claims or write locks.
+ *-----------------------------------------------------------------------------
+ */
 static inline void
 cache_extent_sync(cache *cc, uint64 addr, uint64 *pages_outstanding)
 {
    cc->ops->extent_sync(cc, addr, pages_outstanding);
 }
 
+/*
+ *-----------------------------------------------------------------------------
+ * cache_flush
+ *
+ * Issues writeback for all page in the cache.
+ *
+ * Asserts that there are no pins, read locks, claims or write locks.
+ *-----------------------------------------------------------------------------
+ */
 static inline void
 cache_flush(cache *cc)
 {
    cc->ops->flush(cc);
 }
 
+/*
+ *-----------------------------------------------------------------------------
+ * cache_evict
+ *
+ * Evicts all the pages.
+ * Asserts that there are no pins, read locks, claims or write locks.
+ * Always returns 0.
+ *-----------------------------------------------------------------------------
+ */
 static inline int
 cache_evict(cache *cc, bool ignore_pinned_pages)
 {
    return cc->ops->evict(cc, ignore_pinned_pages);
 }
 
+/*
+ *-----------------------------------------------------------------------------
+ * cache_cleanup
+ *
+ * XXX ?
```

Comment:
Bad naming.

This function performs cache callbacks. Generally this isn't required to be called externally, but there are some cases where it must be. For example, prior to shutdown, calling this ensures that all callbacks are called.